### ISSUE TITLE
training from scratch does not reproduce the results (scannet B15/N4)

### ISSUE THREAD

**Issue by Rabbia-Hassan** [CVMI-Lab/PLA#38]
Hello !
Thank you for your interesting work.
when I trained your model from scratch, (using scannet B15/N4 split) it produces vedy low results upon inference. I am attaching the results for your consideration.
![image](https://github.com/CVMI-Lab/PLA/assets/102951740/1ee82826-bc31-434b-b601-6147a0087729)
While, when I test the pre-trained model provided by you, it gives the following results.
![image](https://github.com/CVMI-Lab/PLA/assets/102951740/d42d351a-b914-4d31-b942-996076e344b8)
That would be great if you can comment on the possible underlying issue here.
Many thanks for your time.
Regards.

**Comment by Dingry** [comment]
It appears there may be an issue with the data. Please ensure that the caption loss is not zero and that the data has been correctly processed, particularly the label for the unannotated class, which should be set to -100 instead of 255.

**Comment by Rabbia-Hassan** [comment]
Thank you very much for the response. But I have used the same data as provided by you. Which should not be the resaon.
Can it possibly be due to fact that I kept the workers=0 in train.py because I has to deal with shared memory issue, while you had kept it to be 4. Do you think keeping it be 0 can cause this problem may be ?
![image](https://github.com/CVMI-Lab/PLA/assets/102951740/15a93dd1-86d7-4458-a2a7-a14b51ae05f2)

many thanks for your time.

**Comment by Rabbia-Hassan** [comment]
Many thanks for the response. Please have a look at the loss curve.
Moreovr, I am giving the link of log file also which might give some insight to error.
https://drive.google.com/file/d/1OvB07eNc7IhKJa4DQSXctxe9sGzYWwCG/view

![image](https://github.com/CVMI-Lab/PLA/assets/102951740/dbd08112-b579-4f47-a6d4-b8a315a819e3)

![image](https://github.com/CVMI-Lab/PLA/assets/102951740/60f10518-c2db-4271-825a-74fbacc13a6f)

![image](https://github.com/CVMI-Lab/PLA/assets/102951740/65b325ea-538d-4449-958b-64b703ebcff5)


![image](https://github.com/CVMI-Lab/PLA/assets/102951740/ad9edaf7-0874-41e5-ae6b-4bba1f8f5f05)


for each epoch, I have computed the average of all the entries to form one value corresponding to each epoch.

**Comment by Rabbia-Hassan** [comment]
please find the link to the complete training log file.

https://drive.google.com/file/d/108eHFp1qROcIBLiHyZ840S3DirpnzIJs/view?usp=sharing

**Comment by Dingry** [comment]
The model's performance before 70 epochs is quite normal. But it suddenly drops to zero at 80 epochs. So weird. We'll check later to see why this happens.

**Comment by jihanyang** [comment]
Could you try to increase the batch size? In our implementation, we use 8 GPUs with 32 total batch sizes. It seems that you are training with a single GPU, so could you try to increase the batch size per GPU?

Just modify the config:
```
OPTIMIZATION:
  BATCH_SIZE_PER_GPU: 4
```
